File: Exercise/Medium/13. Print Number in Large Format/Code/Print Number in Large Format.ts
function PrintLargeDigit(digit: number): void {
    switch (digit) {
        case 0:
            console.log(" ##### ");
            console.log("#     #");
            console.log("#     #");
            console.log("#     #");
            console.log("#     #");
            console.log("#     #");
            console.log(" ##### ");
            break;
        case 1:
            console.log("   #   ");
            console.log("  ##   ");
            console.log("   #   ");
            console.log("   #   ");
            console.log("   #   ");
            console.log("   #   ");
            console.log(" ##### ");
            break;
        case 2:
            console.log(" ##### ");
            console.log("#     #");
            console.log("      #");
            console.log(" ##### ");
            console.log("#      ");
            console.log("#      ");
            console.log("#######");
            break;
        case 3:
            console.log(" ##### ");
            console.log("#     #");
            console.log("      #");
            console.log(" ##### ");
            console.log("      #");
            console.log("#     #");
            console.log(" ##### ");
            break;
        case 4:
            console.log("#      ");
            console.log("#    # ");
            console.log("#    # ");
            console.log("#######");
            console.log("     # ");
            console.log("     # ");
            console.log("     # ");
            break;
        case 5:
            console.log("#######");
            console.log("#      ");
            console.log("#      ");
            console.log(" ##### ");
            console.log("      #");
            console.log("#     #");
            console.log(" ##### ");
            break;
        case 6:
            console.log(" ##### ");
            console.log("#     #");
            console.log("#      ");
            console.log("###### ");
            console.log("#     #");
            console.log("#     #");
            console.log(" ##### ");
            break;
        case 7:
            console.log("#######");
            console.log("     # ");
            console.log("    #  ");
            console.log("   #   ");
            console.log("  #    ");
            console.log(" #     ");
            console.log("#      ");
            break;
        case 8:
            console.log(" ##### ");
            console.log("#     #");
            console.log("#     #");
            console.log(" ##### ");
            console.log("#     #");
            console.log("#     #");
            console.log(" ##### ");
            break;
        case 9:
            console.log(" ##### ");
            console.log("#     #");
            console.log("#     #");
            console.log("###### ");
            console.log("      #");
            console.log("#     #");
            console.log(" ##### ");
            break;
    }
}

function PrintLargeNumber(number: number): void {
    if (number === 0) {
        PrintLargeDigit(0);
        return;
    }

    const digits: number[] = [];
    while (number > 0) {
        digits.push(number % 10);
        number = Math.floor(number / 10);
    }

    for (let i = digits.length - 1; i >= 0; --i) {
        PrintLargeDigit(digits[i]);
    }
}

PrintLargeNumber(12);

File: Exercise/Medium/13. Print Number in Large Format/Code/Print Number in Large Format.cpp
#include <iostream>
#include <vector>

using namespace std;

void printLargeDigit(int digit) {
    switch (digit) {
        case 0:
            cout << " ##### \n";
            cout << "#     #\n";
            cout << "#     #\n";
            cout << "#     #\n";
            cout << "#     #\n";
            cout << "#     #\n";
            cout << " ##### \n";
            break;
        case 1:
            cout << "   #   \n";
            cout << "  ##   \n";
            cout << "   #   \n";
            cout << "   #   \n";
            cout << "   #   \n";
            cout << "   #   \n";
            cout << " ##### \n";
            break;
        case 2:
            cout << " ##### \n";
            cout << "#     #\n";
            cout << "      #\n";
            cout << " ##### \n";
            cout << "#      \n";
            cout << "#      \n";
            cout << "#######\n";
            break;
        case 3:
            cout << " ##### \n";
            cout << "#     #\n";
            cout << "      #\n";
            cout << " ##### \n";
            cout << "      #\n";
            cout << "#     #\n";
            cout << " ##### \n";
            break;
        case 4:
            cout << "#      \n";
            cout << "#    # \n";
            cout << "#    # \n";
            cout << "#######\n";
            cout << "     # \n";
            cout << "     # \n";
            cout << "     # \n";
            break;
        case 5:
            cout << "#######\n";
            cout << "#      \n";
            cout << "#      \n";
            cout << " ##### \n";
            cout << "      #\n";
            cout << "#     #\n";
            cout << " ##### \n";
            break;
        case 6:
            cout << " ##### \n";
            cout << "#     #\n";
            cout << "#      \n";
            cout << "###### \n";
            cout << "#     #\n";
            cout << "#     #\n";
            cout << " ##### \n";
            break;
        case 7:
            cout << "#######\n";
            cout << "     # \n";
            cout << "    #  \n";
            cout << "   #   \n";
            cout << "  #    \n";
            cout << " #     \n";
            cout << "#      \n";
            break;
        case 8:
            cout << " ##### \n";
            cout << "#     #\n";
            cout << "#     #\n";
            cout << " ##### \n";
            cout << "#     #\n";
            cout << "#     #\n";
            cout << " ##### \n";
            break;
        case 9:
            cout << " ##### \n";
            cout << "#     #\n";
            cout << "#     #\n";
            cout << "###### \n";
            cout << "      #\n";
            cout << "#     #\n";
            cout << " ##### \n";
            break;
    }
}

void printLargeNumber(int number) {
    if (number == 0) {
        printLargeDigit(0);
        return;
    }

    vector<int> digits;
    while (number > 0) {
        digits.push_back(number % 10);
        number /= 10;
    }

    for (auto it = digits.rbegin(); it != digits.rend(); ++it) {
        printLargeDigit(*it);
    }
}

int main() {
    int number;

    cin >> number;

    printLargeNumber(number);

    return 0;
}
